### PR TITLE
[v2] Add common components

### PR DESCRIPTION
**Pull request by jotrick**
PR for the common project of the solution

Holds generally available common code

### CODE REVIEW

File: Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/NotificationData/NotificationDataEntity.cs [hunk 1]
```diff
@@ -156,5 +156,20 @@ public IEnumerable<string> Rosters
         /// Gets or sets the sending started DateTime value.
         /// </summary>
         public DateTime? SendingStartedDate { get; set; }
+
+        /// <summary>
+        /// Gets or sets the exception message.
```
Comment (aosolis) [inline Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/NotificationData/NotificationDataEntity.cs:161]: @jotrick Can you elaborate on these? What's an exception message vs a warning message?

File: Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/NotificationData/NotificationDataRepository.cs [hunk 2]
```diff
@@ -123,7 +122,7 @@ public async Task<string> MoveDraftToSentPartitionAsync(NotificationDataEntity d
 
             var newNotificationEntity = new NotificationDataEntity
             {
-                PartitionKey = PartitionKeyNames.NotificationDataTable.DraftNotificationsPartition,
+                PartitionKey = NotificationDataTableNames.DraftNotificationsPartition,
                 RowKey = newId,
                 Id = newId,
                 Title = notificationEntity.Title + " (copy)",
```
Comment (aosolis) [inline Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/NotificationData/NotificationDataRepository.cs:128]: Please put a TODO here to move this string to a resource file

File: Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/SentNotificationData/SentNotificationDataEntity.cs [hunk 3]
```diff
@@ -27,6 +27,14 @@ public class SentNotificationDataEntity : TableEntity
         /// </summary>
         public static readonly string Throttled = "Throttled";
 
+        /// <summary>
+        /// Continued string - this is a state where sending the current notification resulted
+        /// in an exception. Because of this, this string will be stored in the repository
+        /// until a more final state is reached and the notification will be attempted to be
+        /// sent again.
+        /// </summary>
+        public static readonly string Continued = "Continued";
```
Comment (aosolis) [inline Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/SentNotificationData/SentNotificationDataEntity.cs:36]: static readonly --> const

File: Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/SentNotificationData/SentNotificationDataEntity.cs [hunk 4]
```diff
@@ -53,8 +61,43 @@ public class SentNotificationDataEntity : TableEntity
         public int StatusCode { get; set; }
 
         /// <summary>
-        /// Gets or sets the delivery status
+        /// Gets or sets a string representing all of the status code responses for this recipient.
```
Comment (aosolis) [inline Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/SentNotificationData/SentNotificationDataEntity.cs:64]: Can you add a <remarks> section that describes what this really means (reference our chat). The relationship between this, StatusCode, NumberOfAttemptsToSend, and IsStatusCodeFromCreateConversation is unclear.

File: Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/SentNotificationData/SentNotificationDataEntity.cs [hunk 5]
```diff
@@ -40,7 +48,7 @@ public class SentNotificationDataEntity : TableEntity
         /// <summary>
         /// Gets or sets the sent DateTime.
         /// </summary>
-        public DateTime SentDate { get; set; }
+        public DateTime? SentDate { get; set; }
 
         /// <summary>
         /// Gets or sets a value indicating whether the status code is from the create conversation call.
```
Comment (aosolis) [inline Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/SentNotificationData/SentNotificationDataEntity.cs:54]: "the status code" -- there are multiple status codes in this class now

File: Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/NotificationData/NotificationDataEntity.cs (hunk 1 above)
Comment (jotrick) [reply Source/Microsoft.Teams.Apps.CompanyCommunicator.Common/Repositories/NotificationData/NotificationDataEntity.cs:161]: Will look into/change after completion of this PR.

**Pull request merged**